mmv1/products/compute/NetworkFirewallPolicyRule.yaml

```diff
@@ -0,0 +1,295 @@
+references:
+  guides:
+  api: 'https://cloud.google.com/compute/docs/reference/rest/v1/networkFirewallPolicies/addRule'
+docs:
+id_format: 'projects/{{project}}/global/firewallPolicies/{{firewall_policy}}/rules/{{priority}}'
+base_url: 'projects/{{project}}/global/firewallPolicies/{{firewall_policy}}'
+self_link: 'projects/{{project}}/global/firewallPolicies/{{firewall_policy}}/getRule?priority={{priority}}'
+create_url: 'projects/{{project}}/global/firewallPolicies/{{firewall_policy}}/addRule'
+update_url: 'projects/{{project}}/global/firewallPolicies/{{firewall_policy}}/patchRule?priority={{priority}}'
+update_verb: 'POST'
+delete_url: 'projects/{{project}}/global/firewallPolicies/{{firewall_policy}}/removeRule?priority={{priority}}'
+delete_verb: 'POST'
+legacy_long_form_project: true
+import_format:
+  - 'projects/{{project}}/global/firewallPolicies/{{firewall_policy}}/rules/{{priority}}'
+  - '{{project}}/{{firewall_policy}}/{{priority}}'
+  - '{{firewall_policy}}/{{priority}}'
+timeouts:
+  insert_minutes: 20
+  update_minutes: 20
+  delete_minutes: 20
+async:
+  actions: ['create', 'delete', 'update']
+  type: 'OpAsync'
+  operation:
+    base_url: '{{op_id}}'
+    kind: 'compute#operation'
+    path: 'name'
+    wait_ms: 1000
+  result:
+    path: 'targetLink'
+    resource_inside_response: false
+  error:
+    path: 'error/errors'
+    message: 'message'
+examples:
+  - name: 'network_firewall_policy_rule'
+    primary_resource_id: 'primary'
+    vars:
+      tag_key: 'tagkey'
+      network: 'network'
+      fw_policy: 'policy'
+      address: 'address'
+    test_env_vars:
+      project_name: 'PROJECT_NAME'
+      org_id: 'ORG_ID'
+      service_acct: 'SERVICE_ACCT'
+parameters:
+  - name: 'firewallPolicy'
+    type: ResourceRef
+    description: |
+      The firewall policy of the resource.
+    url_param_only: true
+    required: true
+    diff_suppress_func: 'tpgresource.CompareSelfLinkOrResourceName'
+    resource: 'NetworkFirewallPolicy'
+    imports: 'name'
+properties:
+  - name: 'creationTimestamp'
+    type: String
+    description: |
+      Creation timestamp in RFC3339 text format.
+    output: true
+  - name: 'kind'
+    type: String
+    description: |
+      Type of the resource. Always `compute#firewallPolicyRule` for firewall policy rules
+    output: true
+  - name: 'ruleName'
+    type: String
+    description: |
+      An optional name for the rule. This field is not a unique identifier and can be updated.
+  - name: 'description'
+    type: String
+    description: 'An optional description for this resource.'
+  - name: 'priority'
+    type: Integer
+    description: |
+      An integer indicating the priority of a rule in the list.
+      The priority must be a positive value between 0 and 2147483647.
+      Rules are evaluated from highest to lowest priority where 0 is the highest priority and 2147483647 is the lowest prority.
+    required: true
+  - name: 'match'
+    type: NestedObject
+    immutable: true
```

Comment:
This field should not be immutable because it is not immutable in the DCL

Reply:
@trodge I can remove it but I added the immutable on this field (and the other nested ones) because without it some already existing tests will break. (E.g.: `TestAccComputeNetworkFirewallPolicyRule_update`)
When updating this resource/field from `direction: EGRESS` to `direction: INGRESS` and have one or more `dest_* ` field already filled it returns this error:

```
{
  "error": {
    "code": 400,
    "message": "Invalid value for field 'resource.direction': 'INGRESS'. Should not specify dest_address_groups for ingress direction.",
    "errors": [
      {
        "message": "Invalid value for field 'resource.direction': 'INGRESS'. Should not specify dest_address_groups for ingress direction.",
        "domain": "global",
        "reason": "invalid"
      }
    ]
  }
}
```

Remove this requires that I fix the test, is that ok?

Reply:
Yes, please fix the test in that case. It looks like it was passing because it was recreating the resource, and we want update tests to update in place.